lib/client/mediator-subscribers/read.js

```diff
@@ -23,17 +23,23 @@ module.exports = function subscribeToReadTopic(syncDatasetTopics, datasetManager
     var self = this;
     parameters = parameters || {};
 
-    //Creating the item in the sync store
-    datasetManager.read(parameters.id).then(function(itemRead) {
-      var readDoneTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.READ, CONSTANTS.DONE_PREFIX, parameters.topicUid);
 
-      self.mediator.publish(readDoneTopic, itemRead);
+    //Reading the item in the sync store
+    datasetManager.getMetaData(function(metadata) {
+      datasetManager.read(parameters.id).then(function(itemRead) {
+        var readDoneTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.READ, CONSTANTS.DONE_PREFIX, parameters.topicUid);
+        if (metadata.syncEvents) {
+          itemRead._syncStatus = metadata.syncEvents[itemRead.id];
```

Comment:
Just checked it, it will throw an error if it cant find an object for this id
`ERROR Error: Error: unknown_uid`